cores/esp32/Tone.cpp

```diff
@@ -0,0 +1,22 @@
+#include <Arduino.h>
+#include "esp32-hal-ledc.h"
+
+void noTone(uint8_t _pin){
+  ledcDetachPin(_pin);
+}
+
+// parameters:
+// _pin - pin number which will output the signal
+// frequency - PWM frequency in Hz
+// duration - time in ms - how long will the signal be outputted.
+//   If not provided, or 0 you must manually call noTone to end output
+void tone(uint8_t _pin, unsigned int frequency, unsigned long duration){
+  ledcSetup(0, frequency, 11);
```

Comment:
this function is the same as:
```cpp
void tone(uint8_t _pin, unsigned int frequency, unsigned long duration){
   ledcWriteTone(0, frequency);
   ledcAttachPin(_pin, 0);
   if(duration){
     delay(duration);
     noTone(_pin);
   }
 }
```

Reply:
Because it's a bad idea to hardcode the ledc channel, I would suggest to default it to 0, but expose another method `setToneChannel(channel)` to allow the user to change it, if necessary

Reply:
I agree with that. Hardcoding channel was the part where I did not like to implement the tone as it is in Arduino. But with you suggestion will be the best solution.

Reply:
ok, will do.